Tests/Rules/AvoidPositionalParameters.tests.ps1

```diff
@@ -21,9 +21,28 @@ Describe "AvoidPositionalParameters" {
         It "returns no violations" {
             $noViolations.Count | Should -Be 0
         }
-        
+
         It "returns no violations for DSC configuration" {
             $noViolationsDSC.Count | Should -Be 0
         }
     }
+
+    Context "Function defined and called in script, which has 3 or more positional parameters triggers rule." {
+        It "returns avoid positional parameters violation" {
+            $sb=
+            {
+                Function Foo {
+                param(
+                    [Parameter(Mandatory=$true,Position=1)] $A,
+                    [Parameter(Position=2)]$B,
+                    [Parameter(Position=3)]$C)
+                }
+                Foo "a" "b" "c"
+            }
+            $warnings = Invoke-ScriptAnalyzer -ScriptDefinition "$sb"
+            $warnings.Count | Should -BeGreaterThan 0
```

Comment:
why not `-Be 1`?

Reply:
The problem is that the other violation appears PSAvoidTrailingWhitespace if I write the script like this:
<$sb=
            {
                Function Foo {
                param(
                    [Parameter(Mandatory=$true,Position=1)] $A,
                    [Parameter(Position=2)]$B,
                    [Parameter(Position=3)]$C)
                }
                Foo "a" "b" "c"
            } >

The warning doesn't show up when the script looks like:
<$sb=
            {
                Function Foo {
                param(
                    [Parameter(Mandatory=$true,Position=1)] $A,
                    [Parameter(Position=2)]$B,
                    [Parameter(Position=3)]$C)
                }
                Foo "a" "b" "c"}>

But I don't like its readability. 

I am not sure if PSAvoidTrailingWhitespace is an expected behaviour in here?

Reply:
The trailing whitespace is because of the indentation of the last brace, if you were to put the brace to the very left the rule would start to appear. Therefore this is by design and expected because it would be the same as if the last line in a script had space characters. Asserting against the exact number is more important than minor optical issues. I would be ok to just call `Invoke-ScriptAnalyzer` with `-ExcludeRule PSAvoidTrailingWhitespace` instead as an alternative. You should then also use the `-Be` operators for the other assertions. I proposed a while ago to filter out only the relevant rules using `Invoke-ScriptAnalyzer -ScriptDefintion $scriptDefinition | Where-Object { $_.RuleName -eq 'PSAvoidUsingPositionalParameters' }` but Jim did not like this idea either.

Reply:
I got rid of AvoidTrailingWhitespaces trigger and of testing error message.